### ISSUE TITLE
Gatsby on Linux documentation

### ISSUE THREAD

**Issue by m-allanson** [gatsbyjs/gatsby#4139]
Create a "Setup Gatsby on Linux" page similar to the [Windows docs](https://www.gatsbyjs.org/docs/gatsby-on-windows/).

From  #4127:

- incorporate @KyleAMathews comments about mentioning native compilation + Python preqrequisites.
- And @brainwipe's docs (listed below):

## Prerequisites

Starters may use npm plugins that require [native node extensions](https://github.com/nodejs/node-gyp). For those to build, you will need to install `build-essential` from your package source.

For example, on Debian/Ubuntu:

`sudo apt-get install build-essential`

**Comment by igolden** [comment]
This one started? Up my alley and I could help

**Comment by fk** [comment]
❤️🙏 @igolden Yay! (and no, only what's in #4127)

**Comment by spences10** [comment]
I'm checking for Windows Documentation using Windows Subsystem Linux (WSL) if there isn't any in the works I'd like to contribute what I have done for my [own personal documentation].

Just for info for those that are not aware WSL has been available in Windows 10 via the [Windows Store] since the [Fall Creators Update] there will be a very large portion of developers that could use this instead of the song and dance you have to go through installing `windows-build-tools` and then maybe Visual Studio Community 2015 and Python.

It's basically the Linux install, the documentation goes over using [`n`] in place of `nvm` as `nvm` is known to [slow bash startup on WSL] I point to using [n-install] to get the latest version of node without any permission errors like you get with the official install of node, which isn't `sudo apt get node` as that installs node legacy @`~v4 `

[own personal documentation]: https://blog.scottspence.me/wsl-setup
[windows store]: https://www.microsoft.com/store/productId/9NBLGGH4MSV6
[fall creators update]: https://blogs.msdn.microsoft.com/commandline/2017/10/11/whats-new-in-wsl-in-windows-10-fall-creators-update/
[`n`]: https://github.com/tj/n
[slow bash startup on WSL]: https://github.com/Microsoft/WSL/issues/776#issuecomment-266112578
[n-install]: https://github.com/mklement0/n-install

**Comment by m-allanson** [comment]
@spences10 Good idea @spences10 - I think this could make a good section to add to a `Gatsby on Linux` docs page. Pointing out differences specific to using WSL vs a more traditional flavour of Linux.

Note that `sharp` (used by [gatsby-plugin-sharp](https://www.gatsbyjs.org/packages/gatsby-plugin-sharp/)) now [ships with prebuilt binaries for common platforms](http://sharp.pixelplumbing.com/en/stable/changelog/#v0200-5th-march-2018), so it's much less common for a platform's build tools to be required for Gatsby projects.

**Comment by spences10** [comment]
Hey @m-allanson thanks 👍 

Is there anything you'd like me to contribute or do you have this on your list?

**Comment by m-allanson** [comment]
I'd love if you wanted to create a _Gatsby on Linux_ docs page? Similar to the [_Gatsby on Windows_](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/gatsby-on-windows.md) page. Even if it's some placeholder content followed by your WSL content that would be great.

The WSL section could cover installing Node correctly along with the build tools. I'd skip anything from your own documentation that veers towards personal preferences :)

**Comment by spences10** [comment]
Sure, I'll put something together in a PR, thanks 👍
